pkg/reconciler/route/controller.go

```diff
@@ -51,10 +52,54 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 	// Watch for changes in sub-resources so we can sync accordingly
 	routeInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
 
+	// Watch for any changes to VirtualServices in the kf namespace.
 	vsInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Route")),
-		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+		FilterFunc: FilterWithNamespace("kf"),
```

Comment:
The previous Filter was looking to see the object was "controlled" by a Route. We can't really do this though because each VirtualService is owned by many Routes and an object can't have multiple Owners with the Controller set to true.

That being said, I agree that the Filter should also do the type filtering. I'll update it.